src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/EmptyBlockCheck.java

```diff
@@ -171,6 +171,10 @@ else if (!hasText(leftCurly)) {
                     ast.getText());
             }
         }
+
+        if (ast.getType() == TokenTypes.LITERAL_CASE) {
+            checkForEmptyCase(ast);
+        }
```

Comment:
bad excuse.
please join the logic and CC problems could be resolved by moving code in separate method.